src/execution_context.ts

```diff
@@ -0,0 +1,69 @@
+import * as semver from 'semver';
+import {Request, Response, NextFunction} from 'express';
+import {requriedNodeJsVersion} from './options';
+
+export interface ExeuctionContext {
+  'logging.googleapis.com/labels': {
+    executionId: string;
+  };
+  'logging.googleapis.com/trace'?: string;
+  'logging.googleapis.com/spanId'?: string;
+}
+
+function generateExecutionId() {
+  const timestampPart = Date.now().toString(36).slice(-6);
+  const randomPart = Math.random().toString(36).slice(-6);
+  return timestampPart + randomPart;
+}
+
+let asyncLocalStorage: any;
```

Comment:
Can we do `AsyncLocalStorage | undefined` as the type?

Reply:
Var `asyncLocalStorage` is declared in the scope of module so that it can be shared between function `executionContextMiddleware` and `getCurrentContext`, while `AsyncLocalStorage` is imported dynamically in function `executionContextMiddleware`. I'm new to NodeJS/TypeScript so not pretty sure how to work with dynamic import and if it's desired to have a module variable like this. Wondering if you have any suggestions on a better structure.

Reply:
I am fairly new as well
@matthewrobertson any recommendations on this?

Reply:
Explained offline. asyncHooks has to be imported dynamically.